### PR TITLE
Some improvements to RETURN_RESULT_* macros

### PR DESCRIPTION
**Pull request by msft-Jeyaram**
This will make it's way to develop too.

### CODE REVIEW

File: Frameworks/include/wil/result.h [hunk 1]
```diff
@@ -1144,27 +1144,16 @@ typedef _Return_type_success_(return >= 0) LONG NTSTATUS;
...
+        }                               \
     } while (0, 0)
 
-#define RETURN_RESULT_IF_FAILED(input, result) \
-    do {                                       \
-        if (FAILED(input)) {                   \
-            return result;                     \
-        }                                      \
-    } while (0, 0)
-#define RETURN_RESULT_IF(input, result)        \
```
Comment (rajsesh-msft) [inline Frameworks/include/wil/result.h]: 😮 how did this work?

Comment (msft-Jeyaram) [inline Frameworks/include/wil/result.h]: it was never used

File: Frameworks/include/wil/result.h [hunk 2]
```diff
@@ -1144,27 +1144,16 @@ typedef _Return_type_success_(return >= 0) LONG NTSTATUS;
...
-    do {                                       \
-        if (FAILED(input)) {                   \
-            return result;                     \
-        }                                      \
-    } while (0, 0)
 #define RETURN_FALSE_IF_FAILED(input) RETURN_FALSE_IF(FAILED(input));
+#define RETURN_RESULT_IF_NULL(input, result) RETURN_RESULT_IF(!input, result);
```
Comment (aballway) [inline Frameworks/include/wil/result.h]: nit: reanme to ptr and compare against nullptr

File: Frameworks/include/wil/result.h [hunk 3]
```diff
@@ -1144,27 +1144,16 @@ typedef _Return_type_success_(return >= 0) LONG NTSTATUS;
             return;          \
         }                    \
     } while (0, 0)
-
-#define RETURN_RESULT_IF_NULL(input, result) \
-    do {                                     \
-        if (!input) {                        \
-            return result;                   \
-        }                                    \
+#define RETURN_RESULT_IF(input, result) \
```
Comment (aballway) [inline Frameworks/include/wil/result.h]: nit: rename input to condition to be consistent with other macros

File: Frameworks/include/wil/result.h [hunk 4]
```diff
@@ -1144,27 +1144,16 @@ typedef _Return_type_success_(return >= 0) LONG NTSTATUS;
...
-            return result;                     \
-        }                                      \
-    } while (0, 0)
-#define RETURN_RESULT_IF(input, result)        \
-    do {                                       \
-        if (FAILED(input)) {                   \
-            return result;                     \
-        }                                      \
-    } while (0, 0)
 #define RETURN_FALSE_IF_FAILED(input) RETURN_FALSE_IF(FAILED(input));
```
Comment (aballway) [inline Frameworks/include/wil/result.h]: nit: same here

File: Frameworks/include/wil/result.h [hunk 5]
```diff
@@ -1144,27 +1144,16 @@ typedef _Return_type_success_(return >= 0) LONG NTSTATUS;
...
+#define RETURN_RESULT_IF_NULL(input, result) RETURN_RESULT_IF(!input, result);
+#define RETURN_RESULT_IF_FAILED(input, result) RETURN_RESULT_IF(FAILED(input), result)
```
Comment (aballway) [inline Frameworks/include/wil/result.h]: nit: rename to hr

**Comment by msft-Jeyaram** [comment]
@DHowett-MSFT I'm going to leave it as it is.
If I do the change, I need to change the usage across and it can introduce accidental bugs.
I'm keeping the usage tight here to avoid errors.

**Pull request merged**